lib/facility.rb

```diff
@@ -1,14 +1,113 @@
 class Facility
-  attr_reader :name, :address, :phone, :services
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
+  attr_reader :name, 
+              :address, 
+              :phone, 
+              :services, 
+              :registered_vehicles,
+              :collected_fees,
+              :services
+
+  def initialize(office_details)
+    @name = parse_name(office_details)
+    @address = parse_address(office_details)
+    @phone = parse_phone(office_details)
     @services = []
+    @registered_vehicles = []
+    @collected_fees = 0
+  end
+
+  def register_vehicle(data)
+    if @services.include?('Vehicle Registration')
+    data.registration_date = Date.today
+    @registered_vehicles.push(data)
+      if data.antique? 
+        data.plate_type = :antique
+        @collected_fees += 25
+      elsif
+        data.electric_vehicle? 
+        data.plate_type = :ev
+        @collected_fees += 200
+      else
+        data.plate_type = :regular
+        @collected_fees += 100
+      end
+      @registered_vehicles
+    end
   end
 
-  def add_services(service)
+  def add_service(service)
     @services << service
   end
+
+  def administer_written_test(data)
+    if @services.include?('Written Test')
+      if data.permit? && data.age >= 16
+          data.license_data[:written] = true
+      end
+    end
+      data.license_data[:written]
+  end
+
+  def administer_road_test(data)
+    if @services.include?('Road Test')
+      if data.license_data[:written] == true
+          data.license_data[:license] = true
+      end
+    end
+      data.license_data[:license]
+  end
+
+  def renew_drivers_license(data)
+    if @services.include?('Renew License')
+      if data.license_data[:written] == true 
+          data.license_data[:license] = true
+            data.license_data[:renewed] = true
+      end
+    end
+      data.license_data[:renewed]
+  end
+
+  def parse_name(office_details)
+    if office_details[:name]
+        office_details[:name]
+    elsif office_details[:title] 
+          office_details[:title] 
+    elsif office_details[:office_name]
+          office_details[:office_name]
+    end
+  end
+
+  def parse_address(office_details)
```

Comment:
This method is a bit long. Consider extracting helper methods like `parse_location1` and `parse_location2` to improve readability